generate/generate_root_test.go

```diff
@@ -0,0 +1,332 @@
+// Copyright 2022 Mineiros GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package generate_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/mineiros-io/terramate/errors"
+	. "github.com/mineiros-io/terramate/test/hclwrite/hclutils"
+
+	"github.com/mineiros-io/terramate/generate"
+)
+
+func TestGenerateRootContext(t *testing.T) {
+	testCodeGeneration(t, []testcase{
+		{
+			name: "empty generates empty file",
+			configs: []hclconfig{
+				{
+					path: "/source/file.tm",
+					add: GenerateFile(
+						Labels("/target/empty.txt"),
+						Expr("context", "root"),
+						Str("content", ""),
+					),
+				},
+			},
+			want: []generatedFile{
+				{
+					dir: "/target",
+					files: map[string]fmt.Stringer{
+						"empty.txt": stringer(""),
+					},
+				},
+			},
+			wantReport: generate.Report{
+				Successes: []generate.Result{
+					{
+						Dir:     "/target",
+						Created: []string{"empty.txt"},
+					},
+				},
+			},
+		},
+		{
+			name: "generate_file with false condition generates nothing",
```

Comment:
Maybe add some tests interleaving generate_file with condition = true + condition = false, to test that mixing them works properly (we have some tricky logic around that, so the tests could be beneficial).

Reply:
done